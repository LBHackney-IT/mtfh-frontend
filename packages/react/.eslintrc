{
  "extends": "@hackney/eslint-config",
  "parserOptions": {
    "project": "./tsconfig.json"
  },
  "rules": {
    "prefer-arrow-callback": "off",
    "import/no-extraneous-dependencies": [
      "error",
      {
        "devDependencies": [
          "**/*/rollup.config.js",
          "**/*.test.ts",
          "**/*.test.tsx"
        ]
      }
    ],
    "jsx-a11y/anchor-is-valid": [
      "error",
      {
        "components": ["Link"],
        "aspects": ["invalidHref"]
      }
    ],
    "jsx-a11y/label-has-associated-control": "off"
    // "jsx-a11y/label-has-associated-control": [
    //   2,
    //   {
    //     "labelComponents": ["Label"],
    //     "labelAttributes": ["label"],
    //     "controlComponents": ["Input"],
    //     "assert": "htmlFor",
    //     "depth": 3
    //   }
    // ]
  }
}
