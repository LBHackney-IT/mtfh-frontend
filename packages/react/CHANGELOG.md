# @lbh-mtfh/react

## 1.1.0

### Minor Changes

- 37b7b3e: Initial release

### Patch Changes

- Updated dependencies [37b7b3e]
  - @lbh-mtfh/system@1.1.0
  - @lbh-mtfh/test-utils@1.1.0
