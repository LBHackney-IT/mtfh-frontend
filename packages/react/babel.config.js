const babel = require("../../babel.config")

module.exports = babel
