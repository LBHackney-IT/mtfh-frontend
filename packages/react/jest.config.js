const baseConfig = require("../../jest.config")

baseConfig.setupFilesAfterEnv.push("<rootDir>/jest.setup.ts")

module.exports = {
  ...baseConfig,
}
