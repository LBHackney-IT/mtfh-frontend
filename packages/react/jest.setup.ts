import * as LBHFrontend from "lbh-frontend"

window.LBHFrontend = LBHFrontend
