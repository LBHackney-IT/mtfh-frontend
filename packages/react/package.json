{
  "name": "@hackney/mtfh-react",
  "version": "1.1.1",
  "description": "LBH React Design System for LBH Modern Tools for Housing",
  "author": "https://hackney.gov.uk/",
  "homepage": "https://github.com/LBHackney-IT/mtfh-frontend",
  "license": "MIT",
  "sideEffects": [
    "*.css"
  ],
  "main": "dist/cjs/index.js",
  "module": "dist/esm/index.js",
  "types": "dist/types/index.d.ts",
  "exports": {
    ".": {
      "require": "./dist/cjs/index.js",
      "default": "./dist/esm/index.js"
    },
    "./reset.css": "./dist/reset.css"
  },
  "files": [
    "dist"
  ],
  "repository": {
    "type": "git",
    "url": "git+https://github.com/LBHackney-IT/mtfh-frontend.git",
    "directory": "packages/react"
  },
  "scripts": {
    "test": "jest --passWithNoTests --runInBand",
    "test:ci": "jest --runInBand",
    "lint": "eslint src --ext js,jsx,ts,tsx",
    "build": "concurrently yarn:build:*",
    "build:js": "rollup -c",
    "build:types": "tsc --noEmit false --emitDeclarationOnly --declaration --declarationDir dist/types --excludeFiles **/*.test.ts?x,**/*.stories.tsx,jest.setup.ts"
  },
  "bugs": {
    "url": "https://github.com/LBHackney-IT/mtfh-frontend/issues"
  },
  "devDependencies": {
    "@hackney/eslint-config": "^1.1.1",
    "@rollup/plugin-commonjs": "^20.0.0",
    "@rollup/plugin-node-resolve": "^13.0.4",
    "@rollup/plugin-typescript": "^8.2.5",
    "@testing-library/jest-dom": "^5.14.1",
    "@testing-library/react": "^12.0.0",
    "@testing-library/user-event": "^13.1.9",
    "axe-core": "^4.2.3",
    "globby": "^12.0.2",
    "history": "^4.10.0",
    "jest": "^27.0.6",
    "jest-axe": "^5.0.1",
    "jest-matchmedia-mock": "^1.1.0",
    "node-sass": "^6.0.1",
    "postcss": "^8.3.5",
    "postcss-import": "^14.0.2",
    "rollup": "^2.52.8",
    "rollup-plugin-peer-deps-external": "^2.2.4",
    "rollup-plugin-postcss": "^4.0.0"
  },
  "dependencies": {
    "@hackney/mtfh-system": "^1.0.2",
    "@hackney/mtfh-test-utils": "^1.0.2",
    "@radix-ui/react-polymorphic": "^0.0.12",
    "@reach/dialog": "^0.15.3",
    "@testing-library/react-hooks": "^7.0.1",
    "classnames": "^2.3.1",
    "formik": "^2.2.9",
    "generic-names": "^3.0.0",
    "govuk-frontend": "^3.13.0",
    "lbh-frontend": "^3.6.1",
    "react": "^17.0.2",
    "react-dom": "^17.0.2",
    "react-merge-refs": "^1.1.0",
    "react-router-dom": "^5.2.0",
    "use-breakpoint": "^2.0.1"
  },
  "peerDependencies": {
    "formik": "^2.2.9",
    "react": "^17.0.2",
    "react-dom": "^17.0.2",
    "react-router-dom": "^5.2.0"
  }
}
