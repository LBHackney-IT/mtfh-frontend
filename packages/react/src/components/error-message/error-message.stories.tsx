import React from "react";

import { ErrorMessage } from "./error-message";

export default {
  title: "Error Message",
  component: ErrorMessage,
};

export const Base = () => {
  return <ErrorMessage>Error message about full name goes here</ErrorMessage>;
};
