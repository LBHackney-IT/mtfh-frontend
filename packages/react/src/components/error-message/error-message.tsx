import React, { forwardRef } from "react";
import type * as Polymorphic from "@radix-ui/react-polymorphic";
import cn from "classnames";

import { VisuallyHidden } from "../visually-hidden";
import styles from "./styles.module.scss";

export interface ErrorMessageProps {
  assistiveText?: string;
}

export type ErrorMessageComponent = Polymorphic.ForwardRefComponent<
  "span",
  ErrorMessageProps
>;

export const ErrorMessage: ErrorMessageComponent = forwardRef(function ErrorMessage(
  {
    as: ErrorMessageComp = "span",
    assistiveText = "Error",
    className,
    children,
    ...props
  },
  ref,
) {
  return (
    <ErrorMessageComp
      ref={ref}
      className={cn(styles.govukErrorMessage, styles.lbhErrorMessage, className)}
      {...props}
    >
      {assistiveText && <VisuallyHidden>{assistiveText}:</VisuallyHidden>}
      {children}
    </ErrorMessageComp>
  );
});
