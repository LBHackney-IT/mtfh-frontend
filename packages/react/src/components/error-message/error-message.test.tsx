import React from "react"
import { testA11y } from "@hackney/mtfh-test-utils"
import { render } from "@testing-library/react"

import { ErrorMessage } from "./error-message"

test("it renders the error message", async () => {
  const { container } = render(<ErrorMessage>Error</ErrorMessage>)
  expect(container).toMatchSnapshot()
  await testA11y(container)
})
