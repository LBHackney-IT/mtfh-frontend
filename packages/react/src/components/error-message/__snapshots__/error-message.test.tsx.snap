// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders the error message 1`] = `
<div>
  <span
    class="govukErrorMessage lbhErrorMessage"
  >
    <span
      class="visuallyHidden"
    >
      Error
      :
    </span>
    Error
  </span>
</div>
`;
